File: src/game/entity/enemies/npcs/FastEnemy.java
package game.entity.enemies.npcs;

import game.models.Coordinates;
import game.models.Direction;
import game.ui.panels.game.PitchPanel;
import game.utils.Paths;

import java.util.Arrays;
import java.util.List;

public class FastEnemy extends IntelligentEnemy {
    public FastEnemy() {
        super();
    }

    public FastEnemy(Coordinates coordinates) {
        super(coordinates);
    }

    @Override
    protected String getBasePath() {
        return Paths.getEnemiesFolder() + "/fast_enemy/fast_enemy";
    }

    @Override
    public String[] getCharacterOrientedImages() {
        return new String[]{
                String.format("%s_%s_%d.png", getBasePath(), imageDirection.toString().toLowerCase(), 0),
                String.format("%s_%s_%d.png", getBasePath(), imageDirection.toString().toLowerCase(), 1),
                String.format("%s_%s_%d.png", getBasePath(), imageDirection.toString().toLowerCase(), 2),
                String.format("%s_%s_%d.png", getBasePath(), imageDirection.toString().toLowerCase(), 3),
        };
    }

    @Override
    public float getSpeed() {
        return 2f;
    }

    @Override
    protected List<Direction> getImageDirections() {
        return Arrays.asList(Direction.RIGHT, Direction.LEFT);
    }

    @Override
    public int getSize() {
        return PitchPanel.COMMON_DIVISOR * 2;
    }
}

File: src/game/entity/enemies/npcs/FastEnemy.kt
package game.entity.enemies.npcs

import game.entity.EntityTypes
import game.entity.models.Coordinates
import game.entity.models.Direction
import game.ui.panels.game.PitchPanel
import game.utils.Paths.enemiesFolder

/**
 * An enemy with increased speed multiplier;
 */
class FastEnemy : IntelligentEnemy {
    constructor() : super() {
        hitboxSizeToHeightRatio = 0.527f
    }

    constructor(id: Long) : this() {
        this.id = id
    }

    constructor(coordinates: Coordinates?) : super(coordinates) {
        hitboxSizeToHeightRatio = 0.527f
    }

    override val entitiesAssetsPath: String get() ="$enemiesFolder/fast_enemy/fast_enemy"

    override fun getCharacterOrientedImages(): Array<String> = Array(4) { index ->
        "${entitiesAssetsPath}_${imageDirection.toString().lowercase()}_$index.png"
    }

    override fun getSpeed(): Float = 1f

    override fun getImageDirections(): List<Direction> = listOf(Direction.RIGHT, Direction.LEFT)

    override val size: Int
        get() = PitchPanel.COMMON_DIVISOR * 2

    override val type: EntityTypes
        get() = EntityTypes.FastEnemy
}